share/osutil/file_linux.go

```diff
@@ -42,14 +43,72 @@ type FileInfoExt struct {
 	UserAdded   bool
 }
 
+func extractFilePath(symlinkPath, errorMessage, delimiter string) string {
+	// Split the error message into two parts 
+	// TODO => make sure we get the file out only if the read link
+	parts := strings.Split(errorMessage, delimiter)
+	if len(parts) < 2 {
+		return ""
+	}
+
+	// Extract the command and file path part
+	commandFilePath := parts[0]
+
+	// Further split by space and take the last element
+	pathParts := strings.Split(commandFilePath, " ")
+	if len(pathParts) == 0 {
+		return ""
+	}
+
+	resultPath := pathParts[len(pathParts)-1]
+	relativePath, err := filepath.Rel(filepath.Dir(symlinkPath), resultPath)
+
+    if err != nil {
+        fmt.Println("Error:", err)
+        return ""
+    }
+
+	return relativePath
+}
```

Comment:
It's not a good idea to depend on error message because it relies on c-runtime and locale.  There is no guarantee that c runtime will use exactly the same error message format, so you may get different message from time to time.